<!DOCTYPE html>
<html lang="en"><head>
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<title>SARS-CoV-2-Variants by @3dgiordano</title>
<meta name="generator" content="Jekyll v3.9.0" />
<meta property="og:title" content="SARS-CoV-2-Variants" />
<meta name="author" content="David Giordano" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="SARS-CoV-2-Variants by @3dgiordano" />
<meta property="og:description" content="SARS-CoV-2-Variants by @3dgiordano" />
<link rel="canonical" href="https://3dgiordano.github.io/SARS-CoV-2-Variants/" />
<meta property="og:url" content="https://3dgiordano.github.io/SARS-CoV-2-Variants/" />
<meta property="og:site_name" content="SARS-CoV-2-Variants by @3dgiordano" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="SARS-CoV-2-Variants by @3dgiordano" />
<script type="application/ld+json">
{"headline":"SARS-CoV-2-Variants by @3dgiordano","description":"SARS-CoV-2 Variants by @3dgiordano","url":"https://3dgiordano.github.io/SARS-CoV-2-Variants/","@type":"WebSite","name":"SARS-CoV-2-Variants by @3dgiordano","author":{"@type":"Person","name":"David Giordano"},"@context":"https://schema.org"}
</script>
<style>
body {
    font-family: Roboto,sans-serif;
    font-style: normal;
    margin: 0;
	padding: 0px;
}
h1 {
	padding: 5px;
	margin: 10px;
}
p {
	padding: 15px;
	margin: 5px;
}

@media screen and (max-width: 1024px) {
       table {
           width:100%;
       }
       tbody td {
           display: block;
		   width:100% !important;
       }
       tbody td:before {
           content: attr(data-th);
           display: block;
		   width:100% !important;
           
       }
}

</style>
<script src='https://cdnjs.cloudflare.com/ajax/libs/apexcharts/3.27.3/apexcharts.min.js'></script>
<script src='https://gmousse.github.io/dataframe-js/dist/dataframe.min.js'></script>

<script id="rendered-js" crossorigin="anonymous" >

function get_color_columns(columns, variants) {
	var colors = [	
	'#9edae5', '#17becf', '#dbdb8d', '#bcbd22', '#c7c7c7', '#7f7f7f', '#f7b6d2', '#e377c2', '#c49c94', '#8c564b', '#c5b0d5', '#9467bd', '#ff9896', '#d62728', '#98df8a', '#2ca02c', '#ffbb78', '#ff7f0e', '#aec7e8', '#1f77b4',
	'#008FFB', '#00E396', '#FEB019', '#FF4560', '#775DD0',
	'#3f51b5', '#03a9f4', '#4caf50', '#f9ce1d', '#FF9800',
	'#33b2df', '#546E7A', '#d4526e', '#13d8aa', '#A5978B',
	'#4ecdc4', '#c7f464', '#81D4FA', '#fd6a6a', '#546E7A',
	'#2b908f', '#f9a3a4', '#90ee7e', '#fa4443', '#69d2e7',
	'#449DD1', '#F86624', '#EA3546', '#662E9B', '#C5D86D',
	'#D7263D', '#1B998B', '#2E294E', '#F46036', '#E2C044',
	'#662E9B', '#F86624', '#F9C80E', '#EA3546', '#43BCCD',
	'#5C4742', '#A5978B', '#8D5B4C', '#5A2A27', '#C4BBAF',
	'#A300D6', '#7D02EB', '#5653FE', '#2983FF', '#00B1F2'
	];
		
	if (typeof variants === 'undefined') {
		// console.log("Uppss variants disapears...");
		return colors;
	}
	
	var palettte = []
	// console.log(variants);
	columns.forEach( variant => {
		
		variant = variant.split("(")[0].trim();
		// console.log(variant);
		if (variant == "Other") {
			palettte.push("#8B8B8B");
		} else {
			var found = variants.indexOf(variant);
			// console.log(found);
			if (found > -1) {
				if (found < colors.length) {
					// console.log(variant + ":" + colors[found]);
					palettte.push(colors[found]);
				} else {
					// console.log(variant + ":OUT");
					palettte.push("#1B1BFF");
				}
			} else {
				// Generate color based on the letters
				if (variant.length == 2) {
					
					var calt = "#BB" + parseInt(((variant[0].charCodeAt(0))* 255) / 131).toString(16) + "" + parseInt(((variant[1].charCodeAt(0))* 255) / 131).toString(16);
					// console.log(variant + ":ALT:" + calt);
					palettte.push(calt);
				} else {
					var calt = "#" + parseInt(((variant[0].charCodeAt(0))* 255) / 131).toString(16) + "8BBB";
					// console.log(variant + ":ALT2:" + calt);
					palettte.push(calt);
				}
			}
		}
	});
	// console.log(palettte);
	return palettte;
}

function renderChart(data_title, data_url, element_selector, variants) {

dfjs.DataFrame.fromCSV(data_url).then(df => {
	var series = [];
	var columns = []
	df.listColumns().forEach(column => {
		if ((['location', 'date']).indexOf(column) == -1 && df.toArray(column).reduce(function(pv, cv) { return parseFloat(pv.toString().replace(',', '.')) + parseFloat(cv.toString().replace(',', '.')); }, 0) > 0) {
			columns.push(column);
			series.push(
				{name:column, data:df.toArray(column).map(function (x) { return parseFloat(x.toString().replace(',', '.'))})}
			);
		};
	}
	);
	
	var dates = df.toArray("date");

	var colors_columns = get_color_columns(columns, variants);

	var options = {
		title: {
		text: data_title,
		align: 'center',
		margi: 0,
		offsetY: 20,
		style: {
			fontSize: '16px',
			fontFamily: "Verdana"
		},
		},
	 colors: colors_columns,
	  chart: {
		type: 'area',
		height: '400px' ,
		width: '640px' ,
		redrawOnParentResize: false,
		stacked: true,
		animations: {enabled: false},
		offsetY: 0,
		toolbar: {
			show: false,},
		 zoom: {
		 enabled: false,},
		},
	noData: {
  text: "No Data",
	},

	legend: {
      show: false},
	dataLabels: {
          enabled: false,
        },
	tooltip: {
		onDatasetHover: { highlightDataSeries: true },
			
		style: {
			fontSize: '9px',
			fontFamily: "Verdana"
		},
		
		y: {
		
			formatter: function(value, { series, seriesIndex, dataPointIndex, w }) {
			  return Number(value).toFixed(0) + "%";
			}
		}
	},
	fill: {
		type: 'solid'
	},
	  stroke: {
			  curve: 'smooth'
			},
	  series: series,
	  
	  xaxis: {
		tooltip: {
          enabled: false},
        type: 'datetime',
		categories: dates },
		yaxis: {
			max: 100,
			show: false,
		}
	};

	var chart = new ApexCharts(document.querySelector(element_selector), options);

	chart.render();
	
	
	});

}

function load_charts(variants) {
	
dfjs.DataFrame.fromCSV('https://raw.githubusercontent.com/3dgiordano/SARS-CoV-2-Variants/main/data/genomics.csv').then(df => {
	
	df_sequence = df.groupBy('location').aggregate(group => group.stat.sum('num_sequences')).rename('aggregation', 'tot_sequences');
	df_variant = df.groupBy('location', 'variant').aggregate(group => group.stat.sum('num_sequences')).rename('aggregation', 'tot_sequences');
	
	// df_sequence.show();
	// df_variant.show();
	
	locations = [...new Set(df.toArray("location"))];
	
	var content_table = document.querySelector("#content");
	var i = 0; 
	var row;
	locations.forEach(location => {
		// console.log(location);
		tot_sequences = df_sequence.filter(row => row.get('location') == location).toArray("tot_sequences");
		tot_variants = df_variant.filter(row => row.get('location') == location).toArray("variant").length;
		
		if (tot_sequences > 50 && tot_variants >= 2) {
				
			if (i % 2 == 0) {
				row = content_table.insertRow();
			}
			var cell = row.insertCell();
			cell.style="width: 50%;";
			cell.id = "location_" + i;
			// cell.innerHTML = location;
			new Promise( () => {
			setTimeout( renderChart(location, 'https://raw.githubusercontent.com/3dgiordano/SARS-CoV-2-Variants/main/data/' + location.replace(' ', '%20') + ".csv", "#" + cell.id, variants) , 0 );
			});
			
			i +=1;
		}
	});
	
});
}

</script>
	
<div style="margin: 20px;">
<h1>SARS-CoV-2 Variants</h1>

<div style="margin: 20px;">

This site tries to give a glimpse of the different variants of SAR-CoV-2 in the world.<br/>
<br/>
<b>Naming:</b><br/>
<br/>
The naming system used is a mix between the names denominated by WHO, the Pango Lineage system and some Nextstrain Clade names.<br/>
<br/>
The project use the denomination of:<br/>
<ul>
<li>WHO/CDC/ECDC/PHE Variants of Concern (VOC)</li>
<li>WHO/CDC/ECDC Variants of Interest (VOI)</li>
<li>WHO Alerts for Further Monitoring (AFM)</li>
<li>ECDC Variants Under Monitoring (VUM)</li>
<li>PHE Variants Under Investigation (VUI)</li>
</ul>
<br/>
If a major variant is detected, the WHO name is used.<br/>
If it is some other variant to consider not yet considered by WHO, its Pango lineage is used.<br/>
Otherwise, the first letters of the Pango lineage are used.<br/>
If the variant is in WHO/CDC/ECDC/PHE VOI, VUI, VOC, AFM or VUM, the label is included.<br/>
<br/>
For more information about lineages names and classifications<br/>
<ul>
<li>WHO <a href="https://www.who.int/en/activities/tracking-SARS-CoV-2-variants/" target="_blank">Tracking SARS-CoV-2 variants</a></li>
<li>CDC <a href="https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-info.html" target="_blank">SARS-CoV-2 Variant Classifications and Definitions</a></li>
<li>ECDC <a href="https://www.ecdc.europa.eu/en/covid-19/variants-concern" target="_blank">SARS-CoV-2 variants of concern</a></li>
<li>PHE <a href="https://www.gov.uk/government/collections/new-sars-cov-2-variant" target="_blank">COVID-19 (SARS-CoV-2) variants</a></li>
</ul>
<br/>
<b>About:</b><br/>
<br/>
The project, site, published data, and graphics are free and open source.<br/>
A work in progress project created by <a href="https://github.com/3dgiordano" target="_blank">@3dgiordano</a>.<br/>
They can access the source code and data in their GitHub project <a href="https://github.com/3dgiordano/SARS-CoV-2-Variants" target="_blank">https://github.com/3dgiordano/SARS-CoV-2-Variants</a>.</br>
<br/>
<b>Important:</b>
Sequencing data is not a random sample selection and does not represent actual prevalence, it is an estimate.<br/>
<br/>
</div>
</div>

<table id="content" style="width: 100%;border-spacing:40px;" >
</table>

<script>new Promise( () => {

	dfjs.DataFrame.fromCSV("https://raw.githubusercontent.com/3dgiordano/SARS-CoV-2-Variants/main/data/variants.csv").then(df => {
		var variants = df.toArray("label");
		load_charts(variants);
	});
	
	
	});
</script>
</body>
</html>
